main.c

```diff
@@ -246,12 +253,37 @@ int main(int argc, const char *argv[])
 	progname_conv = strdup(path);
 	progname      = basename_internal(progname_conv);
 #endif /* CONFIG_APPELFLOADER_VFSEXEC */
+
 	/* Cut off kernel name (argv[0]) from argument vector */
 	argv = &argv[1];
 	argc -= 1;
 
 #endif /* !CONFIG_APPELFLOADER_CUSTOMAPPNAME */
 
+#if CONFIG_APPELFLOADER_VFSEXEC_ENVPATH
+	env_path = getenv("PATH");
+	if (env_path) {
+		realpath = locate_exec(path, env_path);
+		if (PTR2ERR(realpath) == -EINVAL) {
+			realpath = NULL;
+		} else if (PTRISERR(realpath) && PTR2ERR(realpath) != -EINVAL) {
```

Comment:
Shouls this have `unlikely`?

Reply:
This is not added by me, I only moved it so that the path is computed before `execve()`. Also I'm still not sure what happens if you mix unlikely with other conditions, so I'd rather not add it here.